packages/playwright-core/src/client/tracing.ts

```diff
@@ -51,6 +52,18 @@ export class Tracing extends ChannelOwner<channels.TracingChannel> implements ap
     await this._startCollectingStacks(traceName);
   }
 
+  async group(name: string, options: { location?: { file: string, line?: number, column?: number } } = {}) {
+    if (!options.location) {
+      const filteredStack = filteredStackTrace(captureRawStack());
```

Comment:
Instead of handling the stack manually, we should fall back to the default machinery. That probably means calling [`_wrapApiCall()`](https://github.com/microsoft/playwright/blob/main/packages/playwright-core/src/client/channelOwner.ts#L171) and introducing an extra parameter to pass a custom location to it, instead of taking the stack trace. This way you'll get source collection and other goodies for free.

You would also need to opt-out of `this._isInternalType` check inside to make this particular `tracing.group()` call public. I *think* that making `isInternal` argument be `boolean | undefined` will allow to differentiate explicit `false` to opt-into public call from default `undefined` which aligns with `this._isInternalType`.

Reply:
Actually i do not really understand what you are talking about here, but can it be that this is all irrelevant when using `metadata.location` as i described in my comment before?